pyro/ops/einsum/torch_log.py

```diff
@@ -23,7 +23,7 @@ def einsum(equation, *operands):
 
     inputs, output = equation.split("->")
     if inputs == output:
-        return operands[0][...]  # create a new object
+        return operands[0].clone()  # create a new object
```

Comment:
Could we add a comment here indicating that we avoid shallow copy to avoid a pytorch bug, so somebody doesn't cleverly revert this line in an effort to reduce memory footprint?

Reply:
Maybe this is why ci is failing

Reply:
Done